### PR TITLE
Added tests infrastructure

### PR DESCRIPTION
**Pull request by tomaszrybicki**

### CODE REVIEW

**Comment by mariuszbarczak** [comment]
Fixes #54

File: tests/README [hunk 1]
```diff
@@ -0,0 +1,30 @@
+*** Running tests ***
```
Comment (mariuszbarczak) [inline tests/README]: Maybe we could move it into separated doc? And link to it here.

File: tests/README [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    Test controller requirements:
```
Comment (mariuszbarczak) [inline tests/README]: What's test controller?

File: tests/README [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        - python >= 3.6
+        - python requirements from the file "modules/test-framework/requirements.txt"
+
+    Device Under Test requirements:
+        - packages specified in file "tests/dut_requirements.txt"
+
+    To install the latter type:
+        pip3 install -r requirements.txt
```
Comment (mariuszbarczak) [inline tests/README]: Can we use repo absolute path?

File: tests/README [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    To start tests call the following command from 'tests' directory:
+        python3 -m pytest --dut-config="path/to/config" --log-path="path/to/logs"
+
+    The command executes pytest module. All tests in the working directory are then
+    discovered and executed on the DUT specified in the dut config.
+
+
+*** The DUT config file ***
```
Comment (mariuszbarczak) [inline tests/README]: What's dut?

File: tests/README [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    DUT config file is a yaml file which configures where the tests are run.
+
+    'ip' field should be filled with valid IP string to use remote ssh executor
+    or it should be commented out when user want to execute tests on local machine
+
+    An example of DUT config can be found in "tests/config/example_config.yml"
+
+    WARNING!
+    If no disks are specified in DUT config, the framework formats all non-system drives!
```
Comment (mariuszbarczak) [inline tests/README]: High light it

File: tests/README [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+
+*** The DUT config file ***
+
+    DUT config file is a yaml file which configures where the tests are run.
+
+    'ip' field should be filled with valid IP string to use remote ssh executor
+    or it should be commented out when user want to execute tests on local machine
+
+    An example of DUT config can be found in "tests/config/example_config.yml"
```
Comment (mariuszbarczak) [inline tests/README]: Link to example?

File: tests/config/example_config.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# DUT type should be either "local" or "ssh"
+# if ssh is selected, then ip, user and password fields are required
```
Comment (mariuszbarczak) [inline tests/config/example_config.yml]: Can we support ssh key auth?

File: tests/config/example_config.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+# user: "example_user"
+# password: "example_password"
+# port: "xxxx"
+
+# env: "HTTPS_PROXY=myproxy.example:12345"
+
+working_dir: "/tmp/open-cas-linux/"
+
+disks:
+  - path: "/dev/device_name1" # disk device path
```
Comment (mariuszbarczak) [inline tests/config/example_config.yml]: Do we need other disk configs?

File: tests/config/example_config.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+type: "local"
+
+# ip: "x.x.x.x"
+# user: "example_user"
+# password: "example_password"
+# port: "xxxx"
+
+# env: "HTTPS_PROXY=myproxy.example:12345"
+
+working_dir: "/tmp/open-cas-linux/"
```
Comment (mariuszbarczak) [inline tests/config/example_config.yml]: /tmp/iotrace/

File: tests/conftest.py [hunk 10]
```diff
@@ -0,0 +1,117 @@
+import pytest
+import os
+import sys
+import yaml
+import traceback
+from IPy import IP
+
+# Path to test-framework
+sys.path.append(os.path.join(os.path.dirname(__file__),
```
Comment (mariuszbarczak) [inline tests/conftest.py]: Do we have to use relative path?

File: tests/README (hunk 5 above)
Comment (tomaszrybicki) [reply tests/README]: Removed warning - test-framework behaviour changed

File: tests/conftest.py (hunk 10 above)
Comment (tomaszrybicki) [reply tests/conftest.py]: Added TODO

Comment (tomaszrybicki) [reply tests/conftest.py]: Added TODO

File: tests/config/example_config.yml (hunk 7 above)
Comment (tomaszrybicki) [reply tests/config/example_config.yml]: To do in the future if not possible in test-framework

**Pull request merged**